File: CSharp/cs_EasyMKT-master/EasyMKT/Field.cs
/* Copyright 2017. Bloomberg Finance L.P.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to
 * deal in the Software without restriction, including without limitation the
 * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
 * sell copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:  The above
 * copyright notice and this permission notice shall be included in all copies
 * or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 * IN THE SOFTWARE.
 */

using System.Collections.Generic;

namespace com.bloomberg.mktdata.samples {

    public class Field {

        private string _name;
        private string old_value;
        private string current_value;
        private Fields fields;

        List<NotificationHandler> notificationHandlers = new List<NotificationHandler>();

        internal Field(Fields fields) {
            this.fields = fields;
        }

        internal Field(Fields fields, string name, string value) {
            this.fields = fields;
            this._name = name;
            this.old_value = null;
            this.current_value = value;
        }

        public string Name() {
            return _name;
        }

        public string Value() {
            return this.current_value;
        }

        internal void SetName(string name) {
            this._name = name;
        }

        internal void SetCurrentValue(string value) {
            this.current_value = value;
        }

        internal void CurrentToOld() {
            this.old_value = this.current_value;
        }

        internal FieldChange GetFieldChanged(){

            FieldChange fc = null;

            if (!this.current_value.Equals(this.old_value))
            {
                fc = new FieldChange();
                fc.field = this;
                fc.oldValue = this.old_value;
                fc.newValue = this.current_value;
            }
            return fc;
        }

        public void AddNotificationHandler(NotificationHandler notificationHandler) {
            notificationHandlers.Add(notificationHandler);
        }

        internal void sendNotifications(List<FieldChange> fcl) {
            if (this.notificationHandlers.Count > 0)
            {
                Notification n = new Notification(Notification.NotificationCategory.MKTDATA, Notification.NotificationType.FIELD, this.fields.security, fcl);
                foreach (NotificationHandler nh in notificationHandlers) {
                    if (!n.consume) nh.ProcessNotification(n);
                }
            }
        }
    }
}

File: Python/easyioi-0.1.0/easyioi/field.py
# field.py

from .fieldchange import fieldchange
from .notification import notification

class Field:

	def __init__(self,parent, name="", value=""):
		self.parent = parent
		self.__name = name
		self.__current_value = value
		self.__old_value = ""
		self.notification_handlers = []

	def value(self):
		return self.__current_value

	def name(self):
		return self.__name

	def set_value(self):
		if self.__current_value != vale:
			self.current_to_old
			self.__current_value = value

	def current_to_old(self):
		self.__old_value = self.__current_value

	def get_field_changed(self):

		if self.__old_value != self.__current_value:
			fc = FieldChange(self,self.__old_value,self.__current_value)
			return fc
		else:
			 print("Field NOT changed   Old: " + self.__old_value + "\t New: " + self.__current_value)
			return None

	def add_notification_handler(self,handler):
		self.notification_handlers.append(handler)

	def send_notifications(self,field_change_list):
		print("Send notification...")
		if len(self.notification_handlers) > 0:
			n = Notification(self.parent.owner.get_notification_category(), Notification.NotificationType.FIELD, self.parent.owner, [self.get_field_changed()])

			for nh in self.notification_handlers:
				nh.process_notification(n)


__copyright__ = """
Copyright 2018. Bloomberg Finance L.P.

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to
deal in the Software without restriction, including without limitation the
rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
sell copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:  The above
copyright notice and this permission notice shall be included in all copies
or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
IN THE SOFTWARE.
"""
